packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -102,7 +104,96 @@ exports.sourceNodes = async (
     return
   }
 
-  const drupalFetchActivity = reporter.activityTimer(`Fetch data from Drupal`)
+  fastBuilds = fastBuilds || false
+  if (fastBuilds) {
+    let lastFetched = 0
+    if (
+      store.getState().status.plugins &&
+      store.getState().status.plugins[`gatsby-source-drupal`] &&
+      store.getState().status.plugins[`gatsby-source-drupal`].lastFetched
+    ) {
+      lastFetched = store.getState().status.plugins[`gatsby-source-drupal`]
+        .lastFetched
+    }
+
+    const drupalFetchIncrementalActivity = reporter.activityTimer(
+      `Fetch incremental changes from Drupal`
+    )
+    let requireFullRebuild = false
+
+    drupalFetchIncrementalActivity.start()
+
+    try {
+      // Hit fastbuilds endpoint with the lastFetched date.
+      const data = await axios.get(
+        `${baseUrl}/gatsby-fastbuilds/sync/${lastFetched}`,
+        {
+          auth: basicAuth,
+          headers,
+          params,
+        }
+      )
+
+      if (data.data.status === -1) {
+        reporter.info(`Unable to pull incremental data changes from Drupal`)
+        setPluginStatus({ lastFetched: data.data.timestamp })
+        requireFullRebuild = true
+      } else {
+        // Touch nodes so they are not garbage collected by Gatsby.
+        const existingNodes = getNodes().filter(
```

Comment:
Expensive at scale... :(

If you have a site with a million nodes, this loop runs a million times.